src/routes/validators/tutorValidators.js

```diff
@@ -0,0 +1,117 @@
+const mongoose = require('mongoose');
+const { body, param, validationResult } = require('express-validator');
+
+const mongooseObjectIdValidator = (id) => mongoose.Types.ObjectId.isValid(id); // returns true or false
+
+exports.editProfileValidator = [
+  param('id')
+    .custom((id) => mongooseObjectIdValidator(id))
+    .withMessage('Invalid id param.'),
+
+  body('firstName')
+    .isString()
+    .withMessage('firstName must be a string.')
+    .optional({ nullable: true }),
+
+  body('lastName')
+    .isString()
+    .withMessage('lastName must be a string.')
+    .optional({ nullable: true }),
+
+  body('email')
+    .isString()
+    .withMessage('email must be a string.')
+    .isEmail()
+    .withMessage('Invalid email')
+    .optional({ nullable: true }),
+
+  body('isTutor')
+    .isBoolean()
+    .withMessage('isTutor must be a boolean.')
+    .optional({ nullable: true }),
+
+  body('avatar')
+    .isString()
+    .withMessage('avatar must be a string.')
+    .optional({ nullable: true }),
+
+  body('subjects')
+    .isArray()
+    .withMessage('subjects must be an array.')
```

Comment:
array of what? it would sound better fi you add the type of data should be included in the array